client/src/cordova/plugin/android/java/org/outline/OutlinePlugin.java

```diff
@@ -202,6 +205,17 @@ private void executeAsync(
           final String tunnelId = args.getString(0);
           boolean isActive = isTunnelActive(tunnelId);
           callback.sendPluginResult(new PluginResult(PluginResult.Status.OK, isActive));
+        } else if (Action.FETCH_RESOURCE.is(action)) {
+          final String url = args.getString(0);
+          LOG.fine(String.format(Locale.ROOT, "Fetching resource at %s ...", url));
+          final FetchResourceResult result = Outline.fetchResource(url);
```

Comment:
This will block. Is that a problem? I'm afraid this will freeze the UI, especially bad on timeouts.

We may need a callback mechanism instead.

Reply:
Aha, we both missed [Line 192](https://github.com/Jigsaw-Code/outline-apps/pull/2221/files#diff-509f058f9661f2600b457b0d73100f9e57d98e705b7b192b63a20cebda53df6dR192), 😄. They are already executed in the default threadpool.